### PR TITLE
Bugfixes needed to get EnvelopedData decrypt to work

### PR DESCRIPTION
**Pull request by abgoldberg**
Here are several minor bug fixes that I needed to get decryption with EnvelopedData to work. This was on Chrome with RSA-OAEP (using default parameters), so it's possible the changes need to be generalized.

Also, I'm sure the leading nulls/zeros fix can be cleaned up, but this works and was expedient.  This seems relevant, and suggests the behavior may vary across browsers: https://bugs.chromium.org/p/chromium/issues/detail?id=383998

### CODE REVIEW

File: src/RSAESOAEPParams.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ export default class RSAESOAEPParams
 			case "pSourceAlgorithm":
 				return new AlgorithmIdentifier({
 					algorithmId: "1.2.840.113549.1.1.9", // id-pSpecified
-					algorithmParams: asn1js.OctetString({ valueHex: (new Uint8Array([0xda, 0x39, 0xa3, 0xee, 0x5e, 0x6b, 0x4b, 0x0d, 0x32, 0x55, 0xbf, 0xef, 0x95, 0x60, 0x18, 0x90, 0xaf, 0xd8, 0x07, 0x09])).buffer }) // SHA-1 hash of empty string
```
Comment (YuryStrozhevsky) [inline src/RSAESOAEPParams.js]: Thanks, this is a real error.

File: src/RSAPrivateKey.js [hunk 2]
```diff
@@ -264,14 +264,14 @@ export default class RSAPrivateKey
...
 		const jwk = {
-			n: toBase64(arrayBufferToString(this.modulus.valueBlock.valueHex), true, true),
-			e: toBase64(arrayBufferToString(this.publicExponent.valueBlock.valueHex), true, true),
-			d: toBase64(arrayBufferToString(this.privateExponent.valueBlock.valueHex), true, true),
-			p: toBase64(arrayBufferToString(this.prime1.valueBlock.valueHex), true, true),
-			q: toBase64(arrayBufferToString(this.prime2.valueBlock.valueHex), true, true),
-			dp: toBase64(arrayBufferToString(this.exponent1.valueBlock.valueHex), true, true),
-			dq: toBase64(arrayBufferToString(this.exponent2.valueBlock.valueHex), true, true),
-			qi: toBase64(arrayBufferToString(this.coefficient.valueBlock.valueHex), true, true)
+                        n: toBase64(arrayBufferToString(this.modulus.valueBlock.valueHex).replace(/^\x00+/, ''), true, true),
```
Comment (YuryStrozhevsky) [inline src/RSAPrivateKey.js]: Please explain why you need this. Or better provide an example of CMS EnvelopedData.

File: src/CryptoEngine.js [hunk 3]
```diff
@@ -173,7 +173,7 @@ export default class CryptoEngine
 									switch(algorithm.hash.name.toUpperCase())
 									{
 										case "SHA-1":
-											jwk.alg = "RSA-OAEP-1";
```
Comment (YuryStrozhevsky) [inline src/CryptoEngine.js]: Where did you find the "RSA-OAEP-1"?

**Comment by YuryStrozhevsky** [comment]
Thanks, some parts of your changes were very usefull. But in order to make CMS Enveloped Data working in ES6 version of PKIjs I need more :) So, please check my changes in latest commit. Also please aware that in order to work 100% correct with this commit you would need to have also latest commit of ASN1js ES6.

File: src/RSAPrivateKey.js (hunk 2 above)
Comment (abgoldberg) [inline src/RSAPrivateKey.js]: I think you'd need an example private key really, and that's hard to share. The reason is that, in my key, for whatever reason (maybe it's what OpenSSL does), the exponent2 (dq) is stored with a leading 0 (zero). When that dq value gets through to WebCrypto, it complained that there were leading zeros. The leading zeros in valueHex translate to null bytes in the string representation, so my fix just strips those off. This change made WebCrypto work fine with my private key (and decrypted successfully). 

See https://chromium.googlesource.com/chromium/src/+/lkgr/components/test/data/webcrypto/bad_rsa_keys.json for a similar test case, and also https://bugs.chromium.org/p/chromium/issues/detail?id=374927 for some of the changes in WebCrypto (chromium) that are responsible for producing an error when there are leading zeros.

File: src/CryptoEngine.js (hunk 3 above)
Comment (abgoldberg) [inline src/CryptoEngine.js]: That was in the previously checked in code. WebCrypto complained with this error:  "The JWK "alg" member was inconsistent with that specified by the Web Crypto call" (which I interpreted to mean that we asked for decryption using RSA-OAEP, so the "alg" parameter needed to also be RSA-OAEP (just like in safari, according to the code just above this)). Changing RSA-OAEP-1 to RSA-OAEP made this error go away.

**Pull request closed without merging**